cpp/src/rolling/rolling.cu

```diff
@@ -183,14 +185,14 @@ process_rolling_window(column_device_view input,
  * @param identity identity value of `InputType`
  */
 template <typename InputType, typename OutputType, typename agg_op, aggregation::Kind op, 
-         int block_size, bool has_nulls, typename WindowIterator>
+         int block_size, bool has_nulls, typename PrecedingWindowIterator, typename FollowingWindowIterator>
```

Comment:
I believe it's because they are transform iterators that are generated using different transform functions
https://github.com/rapidsai/cudf/blob/a7bacb4575afc4d97e52f04dd71ff14eb785828b/cpp/src/rolling/rolling.cu#L698-L699
They might generate the same type of function objects ([Wandbox example](https://wandbox.org/permlink/6amtU4TBPc6W5nng)) so it's worth a try to revert to `WindowIterator` and see.

Reply:
The problem was that the two lambdas had different types, and couldn't be encapsulated with a single `WindowIterator` type.

Reply:
Yep, I just verified that this change is required. The types of the two lambdas don't match, in spite of the matching signatures. 
(I suppose, on second thought, it makes sense that the compiler generated types corresponding to the lambdas don't match. But that wasn't obvious on the first go around. :])